utils/ssh.py

```diff
@@ -197,20 +201,33 @@ def get_transport(self, *args, **kwargs):
             self.connect()
         return super(SSHClient, self).get_transport(*args, **kwargs)
 
-    def run_command(self, command, timeout=RUNCMD_TIMEOUT, reraise=False, ensure_host=False):
+    def run_command(
+            self, command, timeout=RUNCMD_TIMEOUT, reraise=False, ensure_host=False,
+            ensure_user=False):
         if isinstance(command, dict):
             command = version.pick(command)
         logger.info("Running command `%s`", command)
+        command_changed = False
         if self.is_container and not ensure_host:
             command = 'docker exec {} bash -c {}'.format(self._container, quote(
                 'source /etc/default/evm; ' + command))
+            command_changed = True
+
+        if self.username != 'root' and not ensure_user:
+            # We need sudo
+            command = 'sudo bash -c {}'.format(quote(command))
+            command_changed = True
+
+        if command_changed:
             logger.info("Actually running command `%s`", command)
         template = '{}\n'
         command = template.format(command)
 
         output = ''
         try:
             session = self.get_transport().open_session()
+            # We need a pseudo-tty for sudo
+            session.get_pty()
```

Comment:
I have doubts this is a good practise. The commands may detect the pty is present and want input from it, blocking forever. Am I not right?

I think we should rather do something like

    sed -i -e 's/^Defaults\s\+requiretty/# \0/' /etc/sudoers

or rather add to sudoers.d something similar to 

    jenkins ALL=(ALL) NOPASSWD:ALL
    Defaults:jenkins        !requiretty

This can be done in the cloud-init for example.